streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -58,8 +60,14 @@ public void addChild(final ProcessorNode<Void, Void, ?, ?> child) {
     public void init(final InternalProcessorContext<Void, Void> context) {
         super.init(context);
         this.context = context;
-        keySerializer = prepareKeySerializer(keySerializer, context, this.name());
-        valSerializer = prepareValueSerializer(valSerializer, context, this.name());
+        try {
+            keySerializer = prepareKeySerializer(keySerializer, context, this.name());
+            valSerializer = prepareValueSerializer(valSerializer, context, this.name());
+        } catch (final ConfigException e) {
+            throw new ConfigException(String.format("Failed to initialize serdes for sink node %s", name()), e);
```

Comment:
Should we split this up further, and have two try-catch blocks, one for the key, and one for the value, to narrow it down further and add key/value as information to the error message?

Reply:
Good idea, I moved the exception handling in a separate class to share the code between the store classes, with corresponding unit tests